File: WinForms/CS/TreeListMainDemo/CodeExamples/Navigation.cs
namespace Examples {
    using System.Windows.Forms;
    using DevExpress.DXperience.Demos.CodeDemo;
    using DevExpress.DXperience.Demos.CodeDemo.Data;
    using DevExpress.Utils;
    using DevExpress.XtraEditors;
    using DevExpress.XtraTreeList;

    [CodeExampleClass("Keyboard Navigation", "Navigation.cs")]
    public static class Navigation {
        [CodeExampleSetUp]
        public static object[] SetUp(XtraUserControl sampleHost) {
            TreeList treeList = new TreeList();
            treeList.OptionsBehavior.Editable = false;
            treeList.Dock = DockStyle.Fill;
            treeList.Parent = sampleHost;
            treeList.DataSource = SpaceObject.LoadData();
            treeList.Load += (s, e) => {
                treeList.Nodes[0].Expand();
                treeList.Nodes[0].Nodes[0].Expand();
            };
            return new object[] { treeList };
        }
        [CodeExampleTearDown]
        public static void TearDown(XtraUserControl sampleHost) {
            foreach(Control control in sampleHost.Controls)
                control.Dispose();
            sampleHost.Controls.Clear();
        }
        //
        [CodeExampleCase("Expand/collapse nodes with left/right arrow keys")]
        [CodeExampleUnderlineTokens("OptionsNavigation", "AllowExpandCollapseWithArrowKeys"), CodeExampleVersionID(171)]
        public static void AllowExpandCollapseWithArrowKeys(TreeList treeList) {
            treeList.OptionsNavigation.AllowExpandCollapseWithArrowKeys = DefaultBoolean.True;
        }
    }
}

File: WinForms/VB/TreeListMainDemo/CodeExamples/Navigation.vb
Imports System.Windows.Forms
Imports DevExpress.DXperience.Demos.CodeDemo
Imports DevExpress.DXperience.Demos.CodeDemo.Data
Imports DevExpress.Utils
Imports DevExpress.XtraEditors
Imports DevExpress.XtraTreeList

Namespace Examples

    <DevExpress.DXperience.Demos.CodeDemo.CodeExampleClass("Keyboard Navigation", "Navigation.cs")>
    Public Module Navigation

        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleSetUp>
        Public Function SetUp(ByVal sampleHost As DevExpress.XtraEditors.XtraUserControl) As Object()
            Dim treeList As DevExpress.XtraTreeList.TreeList = New DevExpress.XtraTreeList.TreeList()
            treeList.OptionsBehavior.Editable = False
            treeList.Dock = System.Windows.Forms.DockStyle.Fill
            treeList.Parent = sampleHost
            treeList.DataSource = DevExpress.DXperience.Demos.CodeDemo.Data.SpaceObject.LoadData()
            AddHandler treeList.Load, Sub(s, e)
                treeList.Nodes(CInt((0))).Expand()
                treeList.Nodes(CInt((0))).Nodes(CInt((0))).Expand()
            End Sub
            Return New Object() {treeList}
        End Function

        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleTearDown>
        Public Sub TearDown(ByVal sampleHost As DevExpress.XtraEditors.XtraUserControl)
            For Each control As System.Windows.Forms.Control In sampleHost.Controls
                control.Dispose()
            Next

            sampleHost.Controls.Clear()
        End Sub

        '
        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleCase("Expand/collapse nodes with left/right arrow keys")>
        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleUnderlineTokens("OptionsNavigation", "AllowExpandCollapseWithArrowKeys"), DevExpress.DXperience.Demos.CodeDemo.CodeExampleVersionID(171)>
        Public Sub AllowExpandCollapseWithArrowKeys(ByVal treeList As DevExpress.XtraTreeList.TreeList)
            treeList.OptionsNavigation.AllowExpandCollapseWithArrowKeys = DevExpress.Utils.DefaultBoolean.[True]
        End Sub
    End Module
End Namespace
